Man page formatting and grammar improvements + "Fedora Linux" (#763)
systemd/mkosi: mkosi.md

  `openmandriva`) for the image, or additional distribution packages
  to install. Note that all options encoded in this configuration file
  may also be set on the command line, and this file is hence little
  more than a way to make sure invoking `mkosi` without further
  parameters in your *source* tree is enough to get the right image of
  your choice set up.

  Additionally, if a *`mkosi.default.d/`* directory exists, each file in
  it is loaded in the same manner adding/overriding the values
  specified in `mkosi.default`. If `mkosi.default.d/` contains a
  directory named after the distribution being built, each file in
  that directory is also processed.

  The file format is inspired by Windows `.ini` files and supports
  multi-line assignments: any line with initial whitespace is
  considered a continuation line of the line before. Command-line
  arguments, as shown in the help description, have to be included in
  a configuration block (e.g.  "`[Packages]`") corresponding to the
  argument group (e.g. "`Packages`"), and the argument gets converted
  as follows: "`--with-network`" becomes "`WithNetwork=yes`". For
  further details see the table above.

> * **`mkosi.skeleton/`** directory or **`mkosi.skeleton.tar`** archive

Reviewer: Somehow I feel there's an article missing here, e.g. "The `mkosi.skeleton/` directory or `mkosi.skeleton.tar` archive" or "A directory `mkosi.skeleton` or archive `mkosi.skeleton.tar`".

Alternatively the sentences would sound natural if directory and archive are omitted and we would rely on directories being marked by a trailing slash and people knowing that `.tar` is an archive.

The same applies to the other changes where emphasis has been added to the file names, where "script" could be omitted. 

If you agree and have a preference, I can make suggestions throughout.

Author: Thanks for the quick review.
I added "the" in various places and changed "is used" → "may be used" as suggested.
I think this reads OK now, but maybe further instances of ", if it exists," should be added…

>   are used to insert files into the image. The files are copied

Reviewer: "are used" -> "may be used", otherwise it's unclear from the paragraph, whether these are mandatory or not.

  copied, except for `mkosi.builddir/`, `mkosi.cache/` and
  `mkosi.output/`. That said, `.gitignore` is respected if the source
  tree is a `git` checkout. If multiple different images shall be
  built from the same source tree it is essential to exclude their
  output files from this copy operation, as otherwise a version of an
  image built earlier might be included in a later build, which is
  usually not intended. An alternative to excluding these built images
  via `.gitignore` entries is to use the `mkosi.output/` directory
  (see below), which is an easy way to exclude all build artifacts.

  The `$MKOSI_DEFAULT` environment variable will be set inside of this
  script so that you know which `mkosi.default` (if any) was passed
  in.

* The **`mkosi.prepare`** script is invoked directly after the

>   software packages are installed, from within the image context. It

Reviewer: ```suggestion
  software packages are installed, from within the image context, if it exists.. It
```

  is once called for the *development* image (if this is enabled, see
  above) with the "build" command line parameter, right before copying
  the extra tree. It is called a second time for the *final* image
  with the "final" command line parameter. This script has network
  access and may be used to install packages from other sources than
  the distro's package manager (e.g. `pip`, `npm`, ...), after all
  software packages are installed but before the image is cached (if
  incremental mode is enabled). This script is executed within
  `$SRCDIR`. In contrast to a general purpose installation, it is safe
  to install packages to the system (`pip install`, `npm install -g`)
  instead of in `$SRCDIR` itself because the build image is only used
  for a single project and can easily be thrown away and rebuilt so
  there's no risk of conflicting dependencies and no risk of polluting
  the host system.

* The **`mkosi.postinst`** script is invoked as the penultimate step

>   of preparing an image, from within the image context. It is called

Reviewer: ```suggestion
  of preparing an image, from within the image context, if it exists. It is called
```
even though "may" is used further down in the paragraph, I think it's clearer to keep it up front.

  first for the *development* image (if this is enabled, see above)
  with the "build" command line parameter, right before invoking the
  build script. It is called a second time for the *final* image with
  the "final" command line parameter, right before the image is
  considered complete. This script may be used to alter the images
  without any restrictions, after all software packages and built
  sources have been installed. Note that this script is executed
  directly in the image context with the final root directory in
  place, without any `$SRCDIR`/`$DESTDIR` setup.

> * The **`mkosi.finalize`** script is invoked as last step of preparing

Reviewer: ```suggestion
* The **`mkosi.finalize`** script, if it exists, is invoked as last step of preparing
```

  an image, from the host system.  It is once called for the
  *development* image (if this is enabled, see above) with the "build"
  command line parameter, as the last step before invoking the build
  script, after the `mkosi.postinst` script is invoked. It is called
  the second time with the "final" command line parameter as the last
  step before the image is considered complete. The environment
  variable `$BUILDROOT` points to the root directory of the
  installation image. Additional verbs may be added in the future, the
  script should be prepared for that. This script may be used to alter
  the images without any restrictions, after all software packages and
  built sources have been installed. This script is more flexible than
  `mkosi.postinst` in two regards: it has access to the host file
  system so it's easier to copy in additional files or to modify the
  image based on external configuration, and the script is run in the
  host, so it can be used even without emulation even if the image has
  a foreign architecture.

> * The **`mkosi.mksquashfs-tool`** script will be called wherever

Reviewer: ```suggestion
* The **`mkosi.mksquashfs-tool`** script, if it exists, will be called wherever
```

  `mksquashfs` would be called.

* The **`mkosi.nspawn`** nspawn settings file will be copied into the

>   same place as the output image file. This is useful since nspawn

Reviewer: ```suggestion
  same place as the output image file, if it exists. This is useful since nspawn
```

  is not set, and it is up to build script to decide whether to do in
  in-tree or an out-of-tree build, and which build directory to use.

* The **`mkosi.includedir/`** directory, if it exists, is
  automatically used as an out-of-tree include directory for header
  files.  Specifically, it will be mounted in the build container at
  `/usr/include/` when building the build image and when running the
  build script. After building the (cached) build image, this
  directory will contain all the files installed to
  `/usr/include`. Language servers or other tools can use these files
  to provide a better editing experience for developers working on a
  project.

* The **`mkosi.installdir/`** directory, if it exists, is
  automatically used as the install directory. Specifically, this
  directory will be mounted into the container at `/root/dest` when
  running the build script. After running the build script, the
  contents of this directory are installed into the final image. This
  is useful to cache the install step of the build. If used,
  subsequent builds will only have to reinstall files that have
  changed since the previous build.

> * The **`mkosi.rootpw`** file provides the password or hashed password

Reviewer: ```suggestion
* The **`mkosi.rootpw`** file can be used to provide the password or hashed password
```

# REQUIREMENTS

mkosi is packaged for various distributions: Debian, Ubuntu, Arch
Linux, Fedora Linux, OpenMandriva. It is usually easiest to use the

> distribution package.

Reviewer: In Debian mkosi is still at version 5, so one's mileage may vary.

Author: It's a package, and it is easy to install. This text doesn't imply anything else ;]
